infra/provision.go

```diff
@@ -0,0 +1,303 @@
+	if err != nil {
+		return err
+	}
+
+	startupScript := pulumi.String(genStartupScript())
+	instance, err := compute.NewInstance(ctx, moniker+"-instance", &compute.InstanceArgs{
+		Name: pulumi.String(moniker),
+		Labels: pulumi.StringMap{
+			"chain_id": pulumi.String(network.ChainID),
+		},
+		MachineType:            pulumi.String(network.NodeConfig.MachineType),
+		Zone:                   pulumi.String(location.Zone),
+		Hostname:               pulumi.String(nodeHostNames[0]),
+		AllowStoppingForUpdate: pulumi.Bool(true),
+		BootDisk:               bootDisk,
+		MetadataStartupScript:  startupScript,
+		Metadata: pulumi.StringMap{
+			"ssh-keys":               ubuntuPubkey,
+			"block-project-ssh-keys": pulumi.String("true"),
+		},
+		ServiceAccount: &compute.InstanceServiceAccountArgs{
+			Email: serviceAccount.Email,
+			Scopes: pulumi.StringArray{
+				pulumi.String("cloud-platform"),
+			},
+		},
+		NetworkInterfaces: compute.InstanceNetworkInterfaceArray{
+			&compute.InstanceNetworkInterfaceArgs{
+				Network: pulumi.String("default"),
+				AccessConfigs: compute.InstanceNetworkInterfaceAccessConfigArray{
+					compute.InstanceNetworkInterfaceAccessConfigArgs{
+						NatIp:       addrs[0],
+						NetworkTier: pulumi.String("STANDARD"),
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	conn := remote.ConnectionArgs{
+		Host:       addrs.ToStringArrayOutput().Index(pulumi.Int(0)),
+		Port:       pulumi.Float64(22),
+		User:       pulumi.String("ubuntu"),
+		PrivateKey: sshPrivate,
+	}
+
+	startupScriptsComplete, err := remote.NewCommand(
+		ctx,
+		moniker+"-bootstrap-script-wait-until-ready",
+		&remote.CommandArgs{
+			Triggers:   pulumi.Array{startupScript},
+			Connection: conn,
+			Update:     pulumi.String("echo updates disabled..."),
+			Create: pulumi.Sprintf(`
+	             for VARIABLE in 1 2 3 4 5 6 7 8 9 .. N
+	             do
+	               if test -f "/tmp/STARTUP_FINISHED"; then
+	                 exit 0
+	               else
+	                 echo 'System startup script incomplete; sleeping 30 seconds...'
+	                 sleep 45
+	               fi
+	             done
+	
+	             echo 'Machine is not ready or system startup script did not complete (timeout)'
+	             exit 1
+	           `),
+		},
+		pulumi.DependsOn([]pulumi.Resource{instance}),
+		pulumi.Timeouts(&pulumi.CustomTimeouts{Create: "10m"}),
+	)
+	if err != nil {
+		return err
+	}
+
+	// ".cw-relayer"
+	techName := network.RelayerHomeFolderName[1:]
+	relayerSpec := unit.UnitSpec{
+		Name:              techName,
+		Description:       fmt.Sprintf("%s daemon", techName),
+		User:              "blockchain",
+		BinaryInstallPath: fmt.Sprintf("/usr/local/bin/%s", network.LocalRelayerBinary),
+	}
+	relayerUnit := relayerSpec.ToUnit(fmt.Sprintf("/home/ubuntu/%s/relayer-config.toml", network.RelayerHomeFolderName))
+	// set environment for relayer keyring pass
+	environment := map[string]string{
+		"CW_RELAYER_PASS": "PASS",
```

Comment:
If we're creating a new wallet to use for each cw relayer we should also generate a password & put it as a secret in the pulumi config. We should consider these wallets funded & production